sgkit/tests/test_distance.py

```diff
@@ -127,19 +148,25 @@ def test_distance_ndarray() -> None:
 
 
 @pytest.mark.parametrize(
-    "metric, metric_func, dtype",
+    "metric, metric_func, dtype, target",
     [
-        ("euclidean", euclidean, "f8"),
-        ("euclidean", euclidean, "i8"),
-        ("correlation", correlation, "f8"),
-        ("correlation", correlation, "i8"),
+        ("euclidean", euclidean, "f8", "cpu"),
+        ("euclidean", euclidean, "i8", "cpu"),
+        pytest.param("euclidean", euclidean, "f8", "gpu", marks=pytest.mark.gpu),
+        pytest.param("euclidean", euclidean, "i8", "gpu", marks=pytest.mark.gpu),
+        ("correlation", correlation, "f8", "cpu"),
+        ("correlation", correlation, "i8", "cpu"),
+        pytest.param("correlation", correlation, "f8", "gpu", marks=pytest.mark.gpu),
+        pytest.param("correlation", correlation, "i8", "gpu", marks=pytest.mark.gpu),
     ],
 )
 def test_missing_values(
     metric: MetricTypes,
     metric_func: typing.Callable[[ArrayLike, ArrayLike], np.float64],
     dtype: str,
+    target: TargetTypes,
 ) -> None:
+    skip_gpu_tests_if_no_gpu(target)
     x = get_vectors(array_type="np", dtype=dtype)
 
     ri_times = np.random.randint(5, 20)
```

Comment:
I think something like this here would be more readable:

```python
x = np.random.choice(range(-3, 3), size=(30, 30))
if np.issubdtype(dtype, np.floating):
    x = np.where(x < 0, np.nan, x)
```

Reply:
I don't see changes for this one.

Reply:
I see I posted the same diff twice in here: https://github.com/pystatgen/sgkit/pull/498#issuecomment-825735250

This suggestion was the second one, which broken on windows.